symplyphysics/laws/optics/the_formula_of_the_spherical_refractive_surface.py

```diff
@@ -0,0 +1,76 @@
+from sympy import (Eq, solve, S)
+from symplyphysics import (units, Quantity, Symbol, print_expression,
+    validate_input, validate_output, dimensionless, convert_to)
+from symplyphysics.core.expr_comparisons import expr_equals
+
+from symplyphysics.laws.conservation import abbe_invariant_of_two_optical_environments_is_constant as abbe_conservation_law
+
+# Description:
+## The formula of the spherical refractive surface allows you to uniquely determine
+## the position of the image if the position of the object is known and vice versa.
+
+# Law: (n / (-lo)) + (n' / li) = (n' - n) / R
+# Where:
+## n - medium refraction index
+## lo - distance from object to surface
+## li - distance from image to surface
+## R - surface curvature radius
+
+# NOTES:
+## - minus before lo if the origin of the coordinate system is placed at the intersection of the optical axis
+##   and the surface of the lens with the radius lying on this optical axis;
+## - proofs: https://studme.org/341451/matematika_himiya_fizik/prelomlenie_otrazhenie_sveta_sfericheskoy_poverhnosti.
```

Comment:
Somehow it looks like this formula explains the image position of the lens, but it is not totally true. In normal lens there are 2 refractive surfaces, so Abbe's invariant should be applied twice. In this proof the focal point is somewhere inside the lens, so only one refractive surface is being calculated. Please mention this fact somewhere in the description.